src/stream/src/task/stream_manager.rs

```diff
@@ -283,52 +270,64 @@ impl LocalStreamManager {
 
     /// Force stop all actors on this worker, and then drop their resources.
     pub async fn stop_all_actors(&self) -> StreamResult<()> {
-        self.core.lock().await.stop_all_actors().await;
+        let actor_handles = self.core.lock().drain_actor_handles();
+        for (actor_id, handle) in &actor_handles {
+            tracing::debug!("force stopping actor {}", actor_id);
+            handle.abort();
+        }
+        for (actor_id, handle) in actor_handles {
+            tracing::debug!("join actor {}", actor_id);
+            let result = handle.await;
+            assert!(result.is_ok() || result.unwrap_err().is_cancelled());
+        }
+        self.context.clear_channels();
+        self.context.actor_infos.write().clear();
+        self.core.lock().clear_state();
+        if let Some(m) = self.await_tree_reg.as_ref() {
+            m.lock().clear();
+        }
         self.reset_barrier_manager();
         // Clear shared buffer in storage to release memory
         self.clear_storage_buffer().await;
 
         Ok(())
     }
 
-    pub async fn take_receiver(&self, ids: UpDownActorIds) -> StreamResult<Receiver> {
-        let core = self.core.lock().await;
-        core.context.take_receiver(&ids)
-    }
-
-    pub async fn update_actors(&self, actors: &[stream_plan::StreamActor]) -> StreamResult<()> {
-        let mut core = self.core.lock().await;
+    pub fn update_actors(&self, actors: &[stream_plan::StreamActor]) -> StreamResult<()> {
+        let mut core = self.core.lock();
         core.update_actors(actors)
     }
 
-    /// This function could only be called once during the lifecycle of `LocalStreamManager` for
-    /// now.
-    pub async fn update_actor_info(&self, actor_infos: &[ActorInfo]) -> StreamResult<()> {
-        let mut core = self.core.lock().await;
-        core.update_actor_info(actor_infos)
-    }
-
     /// This function could only be called once during the lifecycle of `LocalStreamManager` for
     /// now.
     pub async fn build_actors(
```

Comment:
So `build` is `build_impl` + `spawn`. Can we assign them clearer names?

Reply:
I've changed the `build_actors_impl` to `create_actors`.